docs/source/conf.py

```diff
@@ -59,3 +63,39 @@
 # relative to this directory. They are copied after the builtin static files,
 # so a file named "default.css" will overwrite the builtin "default.css".
 html_static_path = []
+
+# `pydoctor_cwd` is the working directory used to call pydoctor.
+# `pydoctor_args` is the list of arguments used to call pydoctor when
+# generating the API docs via the Sphinx pydoctor extension.
+#
+# The following placeholders are available are resolved at runtime:
+# * `{outdir}` the Sphinx output dir
+#
+# Any output from pydoctor is converted into a Sphinx warning.
+#
+_git_reference = subprocess.getoutput('git rev-parse --abbrev-ref HEAD')
+if _git_reference == 'HEAD':
+    # It looks like the branch has no name.
+    # Fallback to commit ID.
+    _git_reference = subprocess.getoutput('git rev-parse HEAD')
+
+if os.environ.get('READTHEDOCS', '') == 'True':
+    rtd_version = os.environ.get('READTHEDOCS_VERSION', '')
+    if '.' in rtd_version:
+        # It looks like we have a tag build.
+        _git_reference = rtd_version
+
+
+pydoctor_cwd = str(pathlib.Path(__file__).parent.parent.parent)
+pydoctor_args = [
+    '--quiet',
+    '--add-package=pydoctor',
+    '--project-name=pydoctor',
+    '--project-url=https://github.com/twisted/pydoctor/',
+    '--docformat=epytext',
+    '--intersphinx=https://docs.python.org/3/objects.inv',
+    '--make-html',
+    '--html-viewsource-base=https://github.com/twisted/pydoctor/tree/' + _git_reference,
+    '--html-output={outdir}/api',
+    '--project-base-dir=' + pydoctor_cwd,
```

Comment:
```suggestion
    f'--project-base-dir={pydoctor_root}',
```
The f-string automatically converts the `Path` (or any other value) to `str`.

Reply:
Old habits die hard....